rainforest/branches.go

```diff
@@ -1,11 +1,65 @@
 package rainforest
 
+import (
+	"fmt"
+	"strconv"
+)
+
 // Branch is a struct representing a Rainforest Branch
 type Branch struct {
 	ID   int    `json:"id,omitempty"`
 	Name string `json:"name"`
 }
 
+// GetBranches returns all branches optionally filtered by name
+func (c *Client) GetBranches(params ...string) ([]Branch, error) {
+	name := ""
+	branches := []Branch{}
+	page := 1
+
+	if len(params) > 0 {
+		name = params[0]
+	}
+
+	for {
+		branchesURL := "branches?page_size=50&page=" + strconv.Itoa(page)
+
+		if name != "" {
+			branchesURL = branchesURL + "&name=" + name
+		}
+
+		req, err := c.NewRequest("GET", branchesURL, nil)
+
+		if err != nil {
+			return nil, err
+		}
+
+		var branchResp []Branch
+		_, err = c.Do(req, &branchResp)
+		if err != nil {
+			return nil, err
+		}
+
+		branches = append(branches, branchResp...)
+
+		totalPagesHeader := c.LastResponseHeaders.Get("X-Total-Pages")
```

Comment:
Can we DRY up this auto-paging functionality?

Reply:
Yep but I won't do it as part of this PR. Added a [ticket](https://rainforest.atlassian.net/browse/RF-25651).